### PR TITLE
blockdev_mirror_complete_running_job: Complete a running mirror job

### PR DESCRIPTION
**Pull request by zhencliu**
Signed-off-by: Zhenchao Liu <zhencliu@redhat.com>

ID: 1934051

### CODE REVIEW

**Comment by zhencliu** [comment]
local fs:
 (1/1) Host_RHEL.m8.u3.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.4.0.x86_64.io-github-autotest-qemu.blockdev_mirror_complete_running_job.q35: PASS (94.63 s)
RESULTS    : PASS 1 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB HTML   : /home/kar/workspace/job-results/job-2021-03-02T06.19-688172f/results.html
JOB TIME   : 97.37 s

nbd:
 (1/1) Host_RHEL.m8.u3.nbd.raw.virtio_scsi.up.virtio_net.Guest.RHEL.8.4.0.x86_64.io-github-autotest-qemu.blockdev_mirror_complete_running_job: PASS (127.53 s)
RESULTS    : PASS 1 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB HTML   : /home/kar/workspace/job-results/job-2021-03-02T06.26-66571f0/results.html
JOB TIME   : 130.26 s

iscsi direct+virtio blk:
 (1/1) Host_RHEL.m8.u3.iscsi_direct.qcow2.virtio_blk.up.virtio_net.Guest.RHEL.8.4.0.x86_64.io-github-autotest-qemu.blockdev_mirror_complete_running_job.q35: PASS (101.96 s)
RESULTS    : PASS 1 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB HTML   : /home/kar/workspace/job-results/job-2021-03-02T06.32-80b1aa0/results.html
JOB TIME   : 104.71 s

ceph:
 (1/1) Host_RHEL.m8.u3.ceph.raw.virtio_scsi.up.virtio_net.Guest.RHEL.8.4.0.x86_64.io-github-autotest-qemu.blockdev_mirror_complete_running_job.q35: PASS (132.79 s)
RESULTS    : PASS 1 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB HTML   : /home/kar/workspace/job-results/job-2021-03-02T06.43-4b66ac2/results.html
JOB TIME   : 135.70 s

**Comment by zhencliu** [comment]
hi @ngu-niny @YongxueHong , can you please help to review this patch? Thanks.

**Review by YongxueHong — APPROVED** [review]
LGTM.

File: qemu/tests/blockdev_mirror_complete_running_job.py [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+
+from provider.blockdev_mirror_nowait import BlockdevMirrorNowaitTest
+
+
+class BlockdevMirrorCompleteRunningJobTest(BlockdevMirrorNowaitTest):
+    """
+    Complete a running blockdev-mirror job
+    """
+
+    def cancel_running_mirror_job(self):
```
Comment (ngu-niny) [inline qemu/tests/blockdev_mirror_complete_running_job.py:11]: Suggest to change the function name to 'complete_running_mirror_job'.

File: qemu/tests/blockdev_mirror_complete_running_job.py [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+    """
+    Complete a running blockdev-mirror job
+
+    test steps:
+        1. boot VM with 2G data disk
+        2. format the data disk and mount it
+        3. create a file
+        4. add a local fs image for mirror to VM via qmp commands
+        5. do blockdev-mirror
+        6. cancel the job when it's running, it should fail
```
Comment (ngu-niny) [inline qemu/tests/blockdev_mirror_complete_running_job.py:41]: cancel -> complete

File: qemu/tests/blockdev_mirror_complete_running_job.py (hunk 1 above)
Comment (zhencliu) [reply qemu/tests/blockdev_mirror_complete_running_job.py]: done

File: qemu/tests/blockdev_mirror_complete_running_job.py (hunk 2 above)
Comment (zhencliu) [reply qemu/tests/blockdev_mirror_complete_running_job.py]: done

**Comment by zhencliu** [comment]
Thank you @ngu-niny , please help to make a double check. Thanks.
 (1/1) Host_RHEL.m8.u3.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.4.0.x86_64.io-github-autotest-qemu.blockdev_mirror_complete_running_job.q35: PASS (84.98 s)
RESULTS    : PASS 1 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB HTML   : /home/kar/workspace/job-results/job-2021-03-09T00.44-567172e/results.html
JOB TIME   : 87.51 s

**Comment by ngu-niny** [comment]
(1/1) Host_RHEL.m8.u4.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.4.0.ppc64le.io-github-autotest-qemu.blockdev_mirror_complete_running_job: PASS (112.89 s)

Acked-by: Nini Gu <ngu@redhat.com>

**Pull request merged**